### PR TITLE
Feature: cannot delete if associated as fk

### PR DESCRIPTION
**Pull request by hetjagani**

### CODE REVIEW

File: service/core/action/medium/delete.go [hunk 1]
```diff
@@ -52,6 +53,53 @@ func delete(w http.ResponseWriter, r *http.Request) {
 		return
 	}
```
Comment (shreeharsha-factly) [inline service/core/action/medium/delete.go:55]: Check medium is in category or not

File: service/core/action/format/delete.go [hunk 2]
```diff
@@ -52,6 +52,17 @@ func delete(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	var cnt int
+	config.DB.Model(&model.Post{}).Where(&model.Post{
+		SpaceID:  uint(sID),
+		FormatID: uint(id),
+	}).Count(&cnt)
```
Comment (shreeharsha-factly) [inline service/core/action/format/delete.go]: @monarkatfactly  I don't think  space_id is required in where query.

File: service/core/action/medium/delete.go (hunk 1 above)
Comment (dmonark) [reply service/core/action/medium/delete.go:55]: check medium in rating & claimant also

File: service/core/action/format/delete.go (hunk 2 above)
Comment (dmonark) [reply service/core/action/format/delete.go]: @shreeharsha-factly I think keeping space ID in query is kind off double security

Comment (hetjagani) [reply service/core/action/format/delete.go]: Yes, it narrows down our result to a particular space.

File: service/core/action/medium/delete.go [hunk 3]
```diff
@@ -52,6 +53,68 @@ func delete(w http.ResponseWriter, r *http.Request) {
...
+	config.DB.Model(&factcheckModel.Claimant{}).Where(&factcheckModel.Claimant{
+		SpaceID:  uint(sID),
+		MediumID: uint(id),
+	}).Count(&totAssociated)
+
+	if totAssociated != 0 {
+		errorx.Render(w, errorx.Parser(util.CannotDeleteError()))
+		return
+	}
+
```
Comment (hetjagani) [inline service/core/action/medium/delete.go]: Added all the associations for medium.

File: service/core/action/format/delete.go (hunk 2 above)
Comment (dmonark) [reply service/core/action/format/delete.go]: @hetjagani check without spaceID if query time decrease then remove it

**Comment by hetjagani** [comment]
Removed the space_id from where queries.

**Review by shreeharsha-factly — APPROVED** [review]

**Pull request merged**